File: src/main/java/com/npg418/uti_afac/util/TooltipAppender.java
package com.npg418.uti_afac.util;

import bre2el.uti.LanguageMapProxy;
import bre2el.uti.TranslatableContentsHelper;
import bre2el.uti.config.Config;
import net.minecraft.network.chat.Component;

import java.util.List;
import java.util.function.Supplier;

public class TooltipAppender {
    public static void appendTooltip(List<Component> tooltips , Supplier<Component> newName) {
        if (LanguageMapProxy.isReady() && Config.dispBothLanguagesOnTooltip) {
            Component currentLine = tooltips.getFirst();
            Component newLine = TranslatableContentsHelper.deepCopyTranslatable(currentLine);
            TranslatableContentsHelper.setChgLangToTextCompo(newLine, !Config.replaceItemNames);
            if (!currentLine.getString().equals(newLine.getString())) {
                addSubLangTitle(tooltips, newLine);
            } else {
                LanguageMapProxy.switchLangMap(true);
                newLine = newName.get();
                LanguageMapProxy.switchLangMap(false);
                if (!currentLine.getString().equals(newLine.getString())) {
                    addSubLangTitle(tooltips, newLine);
                }
            }
        }
    }

    private static void addSubLangTitle(List<Component> tooltips, Component newLine) {
        int rowIndex = Config.replaceItemNames == Config.subLangOn1stLine ? 1 : 0;
        tooltips.add(rowIndex, newLine);
    }
}

File: src/main/java/com/npg418/uti_afac/util/TooltipAppender.kt
package com.npg418.uti_afac.util

import bre2el.uti.LanguageMapProxy
import bre2el.uti.TranslationTextComponentHelper
import bre2el.uti.config.Config
import net.minecraft.network.chat.Component

fun appendTooltip(tooltips: MutableList<Component>, newName: () -> Component) {
    if (LanguageMapProxy.isReady() && Config.dispBothLanguagesOnTooltip) {
        val currentLine = tooltips[0]
        var newLine = TranslationTextComponentHelper.deepCopyTranslatable(currentLine)
        TranslationTextComponentHelper.setChgLangToTextCompo(newLine, !Config.replaceItemNames)
        if (currentLine.string != newLine.string) {
            addSubLangTitle(tooltips, newLine)
        } else {
            LanguageMapProxy.cancelLangMapToUs(true)
            newLine = newName()
            LanguageMapProxy.cancelLangMapToUs(false)
            if (currentLine.string != newLine.string) {
                addSubLangTitle(tooltips, newLine)
            }
        }
    }
}

fun addSubLangTitle(textLines: MutableList<Component>, newLine: Component) {
    val rowIndex = if (Config.replaceItemNames == Config.subLangOn1stLine) 0 else 1
    textLines.add(rowIndex, newLine)
}
